crates/oxc_minifier/src/ast_passes/peephole_substitute_alternate_syntax.rs

```diff
@@ -503,6 +504,38 @@ impl<'a, 'b> PeepholeSubstituteAlternateSyntax {
         Some(ctx.ast.move_expression(&mut expr.right))
     }
 
+    /// `a || (b || c);` -> `(a || b) || c;`
+    fn try_rotate_logical_expression(
```

Comment:
TODO: need to improve perf here. what's faster? a loop over recursion.

we could also early-return if the parent is a logical expression